internal/debug/debugger.go

```diff
@@ -0,0 +1,106 @@
+package debug
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+const PROMPT string = "dbg> "
+
+type cmdHandler func(c *cmdContext) error
+
+type cmdContext struct {
+	Cmd  string
+	Args []string
+}
+
+type Debugger interface {
+	Loop() error
+}
+
+type debugger struct {
+	// Will be further populated with data as we implement more features
+
+	// Store commands both in slice and map to avoid
+	// iterating ove map each time we need the commands
+	// e.g. to display help
+	cmds         []string
```

Comment:
Is there really a penalty in iterating over a map's keys?

Reply:
Actually I ended up not using it so let me remove for now.